course2/meeting10/main.cc

```diff
@@ -0,0 +1,54 @@
+#include <random>
+#include <algorithm>
+#include <vector>
+#include <list>
+#include <deque>
+#include <iostream>
+#include <chrono>
+#include <functional>
+
+struct RNG
+{
+	std::default_random_engine randEngine;
+	std::uniform_real_distribution<double> realDist{0, 1000};
+	double operator() () { return realDist(randEngine); }
+};
+
+void PrintTime(std::function<void()> func, const char* message)
```

Comment:
1. I'm not sure I understand this. std::function is itself a template class, right? So std::function<void()> means: a function that returns void and takes no arguments, whereas e.g. std::function<int(double,double)> would mean a function that returns int and takes two double arguments? So I don't understand the (practical?) difference with a template.

2. Ok but the benchmark is taken after the overhead is incurred, in this case :-)